neo/rawio/intanrawio.py

```diff
@@ -53,40 +57,101 @@ class IntanRawIO(BaseRawIO):
+        if not filename.exists() or not filename.is_file():
+            raise FileNotFoundError(f"{filename} does not exist")
+
         if self.filename.endswith(".rhs"):
+            self.file_format = 'header-attached'
             self._global_info, self._ordered_channels, data_dtype, header_size, self._block_size = read_rhs(
                 self.filename
             )
+        # 3 possibilities for rhd files, one combines the header and the data in the same file with suffix `rhd` while
+        # the other two separates the data from the header which is always called `info.rhd`
+        # attached to the actual binary file with data
         elif self.filename.endswith(".rhd"):
-            self._global_info, self._ordered_channels, data_dtype, header_size, self._block_size = read_rhd(
-                self.filename
+            if filename.name == "info.rhd":
+                # first we have one-file-per-signal which is where one neo stream/file is saved as .dat files
+                if any((filename.parent / file).exists() for file in one_file_per_signal_filenames):
+                    self.file_format = 'one-file-per-signal'
+                    raw_file_paths_dict = create_one_file_per_signal_dict(filename.parent)
+                # then there is one-file-per-channel where each channel in a neo stream is in its own .dat file
+                else:
+                    self.file_format = 'one-file-per-channel'
+                    raw_file_paths_dict = create_one_file_per_channel_dict(filename.parent)
+            # finally the format with the header-attached to the binary file as one giant file
+            else:
+                self.file_format = 'header-attached'
+
+            self._global_info, self._ordered_channels, data_dtype, header_size, self._block_size, channel_number_dict = read_rhd(
+                self.filename, self.file_format
             )
 
         # memmap raw data with the complicated structured dtype
-        self._raw_data = np.memmap(self.filename, dtype=data_dtype, mode="r", offset=header_size)
+        # if header-attached there is one giant memory-map
+        if self.file_format == "header-attached":
+            self._raw_data = np.memmap(self.filename, dtype=data_dtype, mode="r", offset=header_size)
+        else:
+            self._raw_data = {}
+            for stream_index, (stream_index_key, stream_datatype) in enumerate(data_dtype.items()):
+                # for 'one-file-per-signal' we have one memory map / neo stream
+                if self.file_format == "one-file-per-signal":
+                    n_chans = channel_number_dict[stream_index_key]
+                    if stream_index_key == 4 or stream_index_key == 5:
+                        n_samples = int(os.path.getsize(raw_file_paths_dict[stream_index_key]) / 2) # uint16 2 bytes
+                    else:
+                        n_samples = int(os.path.getsize(raw_file_paths_dict[stream_index_key]) / (n_chans * 2))# unit16 2 bytes
```

Comment:
Just for your info.

Pathlib objects have a method called stat that allows you to access the size directly without needing to call the `os` module:

https://docs.python.org/3/library/pathlib.html#pathlib.Path.stat

So you could do something like this:
```
    file_path = raw_file_paths_dict[stream_index_key]
    size_in_bytes = file_path.stat().st_size
    size_in_bytes_of_dtype = np.dtype(stream_datatype).itemsize
    n_samples = stream_factor * size_in_bytes // size_in_bytes_of_dtype
```

Reply:
I knew pathlib had one. I just knew the os right away and was too lazy. Thanks though :) I can change that. I like pathlib better in general.

Reply:
I was trying to decide if I wanted to do this now or later. Maybe resolved for now and we can think about improving later?

Reply:
Yeah, we can do it later if you are not fully convinced : )

Reply:
I'm fully convinced I'm just lazy about this one....

Reply:
I can do it after.

Reply:
OK, now in a PR to this branch:
https://github.com/zm711/python-neo/pull/3